lib/Client/AbstractClient.php

```diff
@@ -20,10 +22,16 @@
      */
     private $options;
 
-    public function __construct(array $options = [])
+    /**
+     * @var ResponseFactoryInterface
+     */
+    protected $responseFactory;
+
+    public function __construct(array $options = [], ?ResponseFactoryInterface $messageFactory = null)
```

Comment:
What is the difference between `?ResponseFactoryInterface $messageFactory = null` and `ResponseFactoryInterface $messageFactory = null`?